src/commands/role.ts

```diff
@@ -0,0 +1,117 @@
+import { Collection, GuildMember, Message, Role, Snowflake } from 'discord.js';
+import Command from 'src/libs/command';
+
+type RoleList = {
+    [key: string]: string;
+};
+
+class RoleCMD extends Command {
+    constructor() {
+        super({
+            name: 'role',
+            aliases: [''],
+            description: 'Command to give yourself a role'
+        });
+    }
+
+    // Usage for invalid command
+    private invalidCMD(message: Message): void {
+        const embedObj = {
+            title: 'Role Commands',
+            description: 'Get a user role.',
+            color: '#e53935',
+            fields: [
+                {
+                    name: '**Usage Example: **',
+                    value: '```ab role [role]``````buddie role [role]```'
+                },
+                {
+                    name: '**Flags: **',
+                    value: 'This command accepts the following flags. \n \n'
+                },
+                {
+                    name: '`--multi`',
+                    value: 'Assign multiple role at once. \n'
+                }
+            ]
+        };
+
+        message.channel.send({ embed: embedObj });
+    }
+
+    public execute(message: Message, args: string[]): void {
+        const first = args.shift();
+
+        let reqRoles: string[] = [];
+
+        // Check for flags
+        if (first === '--multi') {
+            if (args.length === 0) {
+                this.invalidCMD(message);
+                return;
+            }
+
+            reqRoles = args;
+        } else {
+            // Single role
+            if (first === undefined) {
+                this.invalidCMD(message);
+                return;
+            }
+
+            reqRoles = [first];
+        }
+
+        const guild = message.guild;
+        const bot = guild?.client.user;
+
+        if (bot === undefined || bot === null) {
+            throw new Error('Bot not found.');
+        }
+
```

Comment:
@rojen11 
The condition can be simplified with the null check but I had a question related to this condition.

**What are the possible scenarios for throwing this error?**

Reply:
![image](https://user-images.githubusercontent.com/44132059/120362205-8c034780-c32a-11eb-907e-b585563195ed.png)
Typescript gave me this error when I didn't check for undefined.

The bot object will be null when a user enters the command in a direct message as the guild will be null.

Reply:
Can't it be fixed with:

```ts
if(!bot) throw new Error("Bot not found.");
```

Thanks for the info regarding the direct message issue. That's really nice and I was really curious about such possibility because this is the first time I am working in Discord bot. :smile:

Reply:
I forgot that was possible.
Thanks

I will push the refactored code.

Reply:
Great :+1:

Reply:
@rojen11  You can refactor other conditions too where you have checked for both `types`.

Reply:
I refactored other conditions where I checked for both types and pushed it in the same commit.
Should I change other single undefined checks to something like !botRole?

Reply:
Those are totally fine and everything else looks good.